modules/gfac/gfac-impl/src/main/java/org/apache/airavata/gfac/impl/task/SCPDataStageTask.java

```diff
@@ -97,7 +100,29 @@ public TaskStatus execute(TaskContext taskContext) {
                         status.setReason("File name is null");
                     }
                     return status;
+
+                }
+
+                if (processOutput.getType() == DataType.URI || processOutput.getType() == DataType.URI_COLLECTION ||
+                        processOutput.getType() == DataType.STDOUT || processOutput.getType() == DataType.STDERR) {
+                    return transferFiles(processContext, subTaskModel, taskContext, processState);
+
+                } else if (processOutput.getType() == DataType.FLOAT || processOutput.getType() == DataType.STRING ||
+                        processOutput.getType() == DataType.INTEGER ) {
+                    return extractStringFromFile(subTaskModel, taskContext, processOutput.getApplicationArgument());
```

Comment:
This seems like a bit of a hack to use applicationArgument this way. Would it not be better to add another property to the OutputDataObjectType model?

Reply:
Do you suggest to add something like readable/ transferrable?

Reply:
I was thinking of something like "outputValuePrefix" to define the string to look for in the file.

Reply:
What could be possible values for outputValuePrefix? And on what condition do you suggest to trigger data extraction?

Reply:
You would keep triggering data extraction based on the output type (FLOAT, STRING, INTEGER). But the string you look for would be specified by outputValuePrefix. Possible values would be null or some string, and it should not be null if the output type is FLOAT, STRING or INTEGER.

The reason I don't think we should use applicationArgument is because it has a different use, it is used to determine the command line, see https://github.com/apache/airavata/blob/433800846fcaa9d254fe598af0958fec9431cc75/modules/gfac/gfac-core/src/main/java/org/apache/airavata/gfac/core/GFacUtils.java#L648-L648.